### PR TITLE
HIVE-27482 - Minor issues in array_intersect udf

### PR DESCRIPTION
**Pull request by tarak271**
### What changes were proposed in this pull request?

1. SELECT array(1,2,3),array_intersect(array(1, 2, 3),array(1,3,4)); gives wrong output

[1,3] [1,3]

instead of 

[1,2,3][1,3]

2.  SELECT array_intersect(array(),array()); should give empty array instead of NULL

3.  Error handling when elements of array are not comparable

These fixes were already present on array_except udf whose implementation is identical to array_intersect, Ref: https://github.com/apache/hive/pull/4090

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

Added qtest files as well as junit test cases

### CODE REVIEW

**Review by okumin — APPROVED** [review]
Looks good! I confirmed all test cases generate the same result as Spark SQL. Thanks for the follow-up! It's really helpful.
We need an approval of a committer because I'm not a committer.

**Comment by okumin** [comment]
It'll be good to rerun CI. I guess the failure is unrelated to this change.

File: ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java
Comment (aturoczy) [inline ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java:60]: I would change it for if (array == null || array2 == null)  {  }  I don't see the value to have separate IF statement

File: ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java
Comment (aturoczy) [inline ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java:68]: If you already touch this code I would suggest to change the retArray3 to **resultArray** or just simple **result** I think it would be more expressive

**Review by aturoczy — CHANGES_REQUESTED** [review]
Minor change request

File: ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java
Comment (tarak271) [reply ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java]: Changed as recommended

File: ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java
Comment (tarak271) [reply ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java]: changed variable names to match their usage

**Review by saihemanth-cloudera — COMMENTED** [review]
LGTM +1. Pending green Ptests. Will approve this PR once it is green.

**Comment by aturoczy** [comment]
There is a test failure, but as I see it should be unrelated/

**Review by saihemanth-cloudera — APPROVED** [review]

**Pull request merged**
